backend/src/process.py

```diff
@@ -203,6 +218,9 @@ async def run_while(
                     "event": "iterator-progress-update",
                     "data": {
                         "percent": 1,
+                        "index": index,
+                        "total": length_estimate,
```

Comment:
This is the case where our length estimate was too large. So `index` contains the actual length.
```suggestion
                        "index": index,
                        "total": index,
```